src/factiva/news/stream/Stream.py

```
@@ -0,0 +1,447 @@
+
+        Raises
+        -------
+        ValueError: snapshot_id and query are undefned
+        '''
+        if not self.snapshot_id and not self.query:
+            raise ValueError('Snapshot id and query not found')
+        if self.snapshot_id:
+            return self._create_by_snapshot_id()
+        else:
+            return self._create_by_query()
+
+    def create_subscription(self):
+        '''
+        Create subcription allows a user
+        to create a another subscription
+        for an existing stream
+
+        Returns
+        -----
+        Dataframe which contains the state about the current stream
+
+        Raises
+        -------
+        RuntimeError: when unable to create a subscription
+        '''
+        try:
+            new_subscription = Subscription(self.stream_id)
+            new_subscription.create(
+                headers=self.stream_user.get_authentication_headers()
+            )
+            self.subscriptions[new_subscription.id] = new_subscription
+            return new_subscription.subscription_id
+        except Exception:
+            raise RuntimeError('Unable to create subscription')
+
+    def delete_subscription(self, id) -> bool:
+        '''
+        Delete subcription allows a user
+        to delete a subscription
+        for an existing stream
+
+        Parameters
+        ----------
+        id :  str
+            is the representation of a given
+            subscription planned to be deleted
+
+        Returns
+        -----
+        Dataframe which contains the state about the current stream
+
+        Raises
+        -------
+        ValueError: when there is invalid subscription id
+        RuntimeError: when unable to delete a subscription
+        '''
+        if id not in self.subscriptions:
+            raise self.INVALID_SUBSCRIPTION_ID_ERROR
+        try:
+            if self.subscriptions[id].delete(
+                headers=self.stream_user.get_authentication_headers()
```

Since the authorization headers needed here are the ones that contain the `user-key`, do you think it's worth to add that validation or we should let the API return the `400` error for missing authentication?

---

I think the api should return the error, we cannot prove by ourselves that they key actually exists